src/headless/utils/parse-helpers.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @copyright 2020, the Converse.js contributors
+ * @license Mozilla Public License (MPLv2)
+ * @description Pure functions to help funcitonally parse messages.
+ * @todo Other parsing helpers can be made more abstract and placed here.
+ */
+const helpers = {
+    // Captures all mentions, but includes a space before the @
+    mentionRegex: /\s([@][\w_-]+)|^([@][\w_-]+)/ig,
+
+    filterIsNicknameMentioned (mentions) {
+        // testing against lowercase nickname to be able to
+        // produce matches that are case insensitive
+        return nickname => mentions.includes(nickname.toLowerCase());
+    },
```

Comment:
I don't see this method being called from anywhere.

Also, it looks quite generic, so we don't have to semantically tie it to nicknames, it could be used to filter for any matching string.

Reply:
I removed the function as you are right I left it unused.